server/models/account/user.ts

```diff
@@ -354,6 +361,91 @@ export class UserModel extends Model<UserModel> {
     return UserModel.findOne(query)
   }
 
+  static loadByLdapDn (ldapDn: string) {
+    const query = {
+      where: {
+        ldapDn
+      }
+    }
+
+    return UserModel.findOne(query)
+  }
+
+  static findOrCreateLDAPUser (username: string) {
+    let userInfos
+
+    return Promise.resolve(UserModel.findLDAPUser(username))
```

Comment:
Use async/await instead

Example: `const { dn ] = await UserModel.findLDAPUser(username)` etc

You can use try/catch, but please log the error